packages/sdk-btc/source/client.service.ts

```diff
@@ -13,15 +13,18 @@ export class ClientService extends Services.AbstractClientService {
 	public override async transactions(
 		query: Services.ClientTransactionsInput,
 	): Promise<Collections.ConfirmedTransactionDataCollection> {
-		let address = query.addresses ? query.addresses[0] : query.address;
 
-		const response = await this.#get(`wallets/${address}/transactions`);
+		if (!query.addresses) {
+			throw new Error("No addresses specified for querying for transactions");
+		}
+
+		const response = await this.#post("wallets/transactions", { addresses: query.addresses });
 
 		return this.dataTransferObjectService.transactions(response.data, this.#createMetaPagination(response));
 	}
 
 	public override async wallet(id: string): Promise<Contracts.WalletData> {
```

Comment:
@faustbrian Here that id should become the wallet's id (whatever it is: priv key, pub key, etc). And we need to do the address derivation so we can query all wallet addresses at once.

Reply:
Should be the same process as ADA, correct?

Reply:
Exactly!

Reply:
With the difference that now we have 3 (or was it 4) addressing schemes. So given the wallet id we should be able to decide (from the DW) what addressing scheme that wallet is using. And then, based on that, we need to derive addresses, up to some point, only that now we don't have graphql for figuring this out... :-(

Reply:
We'll probably know what scheme they want to use based on how some DW designs are turning out.